src/hotspot/share/opto/vectornode.cpp

```diff
@@ -818,22 +834,26 @@ VectorNode* VectorNode::scalar2vector(Node* s, uint vlen, BasicType bt, bool is_
 }
 
 VectorNode* VectorNode::shift_count(int opc, Node* cnt, uint vlen, BasicType bt) {
-  // Match shift count type with shift vector type.
+  int vopc = VectorNode::shift_count_opcode(opc);
   const TypeVect* vt = TypeVect::make(bt, vlen);
+  return VectorNode::make(vopc, cnt, nullptr, vt);
+}
+
+int VectorNode::shift_count_opcode(int opc) {
   switch (opc) {
   case Op_LShiftI:
   case Op_LShiftL:
-    return new LShiftCntVNode(cnt, vt);
+    return Op_LShiftCntV;
   case Op_RShiftI:
   case Op_RShiftL:
   case Op_URShiftB:
   case Op_URShiftS:
   case Op_URShiftI:
   case Op_URShiftL:
-    return new RShiftCntVNode(cnt, vt);
+    return Op_RShiftCntV;
   default:
-    fatal("Missed vector creation for '%s'", NodeClassNames[opc]);
-    return nullptr;
+    fatal("Node class '%s' is not supported for shift count", NodeClassNames[opc]);
+    return -1;
```

Comment:
Note: splitting the old `VectorNode::shift_count` (scalar opc -> L/RShiftCntVNode).
- Created new `VectorNode::shift_count_opcode` (scalar opc -> vector shift cnt opc)
- Moved node creation to `VectorNode::make` (vector shift cnt opc -> L/RShiftCntVNode)

I need this for later VTransform changes, where I need to do the vector-opc determination in one stage (VTransform build), and then the vector-node creation in a later stage (VTransform apply).